### ISSUE TITLE
Plugin style merge issues: Rules appear to be getting overwritten by 2nd class not merged into rules from 1st class

### ISSUE THREAD

**Issue by rluiten** [ben-rogerson/twin.macro#28]
Demo of a problem I am seeing.

I created a plugin for sidebar styling it is incomplete but I was seeing something that surprised me.
I am not certain this is a bug, but it's not doing what I expected, though possibly my expectations may be faulty.

Styles from two classes appear to be getting over written for a given style rule path "> *" rather than merged, so the 2nd classes styles for that path are replace the rules from the first class styles.

https://codesandbox.io/s/cra-tailwind-emotion-starter-demo-possible-style-not-merging-bug-twinmacro-tf4vk

I found if i use the rule `.with-sidebar__rightAlternate` the styles merge but its kind of cheating by adding `> *:first-child` so it doesn't match the rule `> *` used in  `.with-sidebar`.

The plugin creating the styles.
```javascript
function sidebarPlugin({ addUtilities }) {
  addUtilities({
    [`.with-sidebar > *`]: {
      display: 'flex',
      flexWrap: 'wrap'
    }
  })
  addUtilities({
    [`.with-sidebar__right > *`]: {
      flexDirection: 'row-reverse'
    }
  })
  addUtilities({
    [`.with-sidebar > * > *`]: {
      flexGrow: 1
    }
  })
  // last-child is always not-sidebar
  addUtilities({
    [`.with-sidebar > * > *:last-child`]: {
      flexBasis: 0,
      flexGrow: 999,
      minWidth: `calc(50%)`
    }

  addUtilities({
    [`.with-sidebar__rightAlternate > *:first-child`]: {
      flexDirection: 'row-reverse'
    }
  })
  })
}
```

Styles rules from different classes not merging.

tw`with-sidebar with-sidebar__right` styles
This shows style "> *" has lost the flex and wrap settings from with-sidebar. Which ever class name is last overwrites the first, it does not merge.
```
{
  "> *": {
    "flex-direction": "row-reverse"
  },
  "> * > *": {
    "flex-grow": "1"
  },
  "> * > *:last-child": {
    "flex-basis": "0",
    "flex-grow": "999",
    "min-width": "calc(50%)"
  }
}
```

tw`with-sidebar__right with-sidebar` styles
This shows style "> *" has lost the styles from with-sidebar__right. Which ever class name is last overwrites the first, it does not merge.
```
{
  "> *": {
    "display": "flex",
    "flex-wrap": "wrap"
  },
  "> * > *": {
    "flex-grow": "1"
  },
  "> * > *:last-child": {
    "flex-basis": "0",
    "flex-grow": "999",
    "min-width": "calc(50%)"
  }
}
```

tw`with-sidebar` styles
```
{
  "> *": {
    "display": "flex",
    "flex-wrap": "wrap"
  },
  "> * > *": {
    "flex-grow": "1"
  },
  "> * > *:last-child": {
    "flex-basis": "0",
    "flex-grow": "999",
    "min-width": "calc(50%)"
  }
}
```

tw`with-sidebar__right` styles
```
{
  "> *": {
    "flex-direction": "row-reverse"
  }
}
```

**Comment by rluiten** [comment]
I just tested using `css` from `@emotion/core` to compose styles as follows.

```javascript
var withSidebar = tw`with-sidebar`
var withSidebarRight = tw`with-sidebar__right`
var composedSidebar = css`
  ${withSidebar};
  ${withSidebarRight};
`
```

The composedSidebar styles are as I would expect. the combination of both classes.

**Comment by ben-rogerson** [comment]
Thanks for submitting this and for creating the bug reproduction.
That does look like some strange behaviour.
Have you tested `sidebarPlugin` works as you expect in regular `tailwindcss`?

**Comment by rluiten** [comment]
No I haven't the result is very clearly that latter classes are top level field name merging only...

As if  { ...class1Styles, ...class2Styles } is being done in twin.macro.

I tried to dig through twin.macro source but didn't find where you are merging styles...

If you can point me at that or and possible a test  around that are I might be able to do a more complete recursive merge.

**Comment by ben-rogerson** [comment]
No worries - here's where the plugin processing starts https://github.com/ben-rogerson/twin.macro/blob/c75aec5afa093beed5778356a098d28902aa0d9d/src/utils.js#L114

**Comment by rluiten** [comment]
I've had a bit of a look at that code, It looks to me like that's the logic that 
loads the the plugin utilities and runs them to generate the styles into the overall tailwind config.
I don't think I made what I was chasing clear enough.

Given this example.

```html
<div tw="class1 class2">a</div>
```

I was looking for the logic that takes the styles for each class passed to tw
and then merges them together into the output style for the div.

For above example get styles for `class1` and merges in the styles for `class2`.

**Comment by ben-rogerson** [comment]
Take a look at `classListOrdered.reduce`: https://github.com/ben-rogerson/twin.macro/blob/628799bc0edc2183dec8e83ce73a196639a8d8e8/src/getStyles.js#L24

**Comment by ben-rogerson** [comment]
The issue is that the plugin output isn't being merged with the reduce accumulator. I'll add a fix for this next version.

**Comment by ben-rogerson** [comment]
This should be fixed in v1.0.0 🎉

**Comment by rluiten** [comment]
Awesome work thank you.
